System/Vector.h

```diff
@@ -148,15 +148,15 @@ namespace RTE {
 		/// Indicates whether both X and Y components of this Vector are 0.
 		/// </summary>
 		/// <returns>Whether both X and Y components of this Vector are 0.</returns>
-		bool IsZero() const { return m_X == 0 && m_Y == 0; }
+		bool IsZero() const { return fabs(m_X) < m_Tol && fabs(m_Y) < m_Tol; }
```

Comment:
Was the old way causing IsZero to fail because of float precision problems? I like your new way better, it's a great solution, I'm just curious.

Reply:
Yeah, that's what I figured. I agree we may want to bump up the tolerance by an order of magnitude, since it's pretty low at the moment.

Reply:
on second thought, it really depends on how this function, IsZero is actually used.
CC's vectors in general might be much larger but just looking at a few occurrences of the function i'm not sure that it's used for what i had in mind when i wrote this.
whatever IsZero is used for, the current tolerance is low enough to be safe.
to decide if it should be higher would require investigating where and how the function is used.

Reply:
It's used in a lot of places in the code (vs says 96 references) and also in lua scripts. I think it's probably okay to leave as-is, but it'd be great if you could do a little testing with an order of magnitude higher, so we can see if that works well/better.

That said, if you're tired of this stuff and don't want to, that's okay too - just put a todo on it or put an issue in saying we should investigate it in future so it's not forgotten entirely.

Reply:
for now i'll put a todo.
after this gets merged, if i remember, i'll also put in an issue.